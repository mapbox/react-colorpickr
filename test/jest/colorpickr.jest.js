import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as testCases from '../test-cases/colorpickr-test-cases';

describe('Colorpickr', () => {
  let testCase;
  let wrapper;

  describe(testCases.basic.description, () => {
    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = shallow(React.createElement(testCase.component, testCase.props));
    });

    test('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('hex input returns value onChange', () => {
      const mockEvent = {
        target: {
          value: 'eeef'
        }
      };

      wrapper.find('[data-test="hex-input"]').props().onChange(mockEvent);
      expect(toJson(wrapper)).toMatchSnapshot();
      expect(testCase.props.onChange).toHaveBeenCalledTimes(1);
      expect(testCase.props.onChange).toHaveBeenCalledWith({
        h: 0,
        s: 0,
        l: 93,
        r: 238,
        g: 238,
        b: 238,
        a: 1,
        hexInput: true,
        hex: 'eeef',
        mode: 'hsl',
        channel: 'h'
      });
    });

    test('invalid hex input does not fire onChange', () => {
      const mockEvent = {
        target: {
          value: 'eeeff'
        }
      };

      wrapper.find('[data-test="hex-input"]').props().onChange(mockEvent);
      expect(toJson(wrapper)).toMatchSnapshot();
      expect(testCase.props.onChange).toHaveBeenCalledTimes(0);
    });

    test('hex input adjusts value onBlur', () => {
      const mockEvent = {
        target: {
          value: 'eeef'
        }
      };

      wrapper.find('[data-test="hex-input"]').props().onBlur(mockEvent);
      expect(toJson(wrapper)).toMatchSnapshot();
      expect(testCase.props.onChange).toHaveBeenCalledTimes(1);
      expect(testCase.props.onChange).toHaveBeenCalledWith({
        h: 0,
        s: 0,
        l: 93,
        r: 238,
        g: 238,
        b: 238,
        a: 1,
        hexInput: true,
        hex: 'eeeeee',
        mode: 'hsl',
        channel: 'h'
      });
    });

    test('default mode is active', () => {
      const checked = wrapper.find('[data-test="mode-hsl"]').props().checked;
      expect(checked).toEqual(true);
    });

    test('default reset action is present', () => {
      const reset = wrapper.find('[data-test="color-reset"]').exists();
      expect(reset).toBe(true);
    });

    test('overrideValue manually sets a new color', () => {
      wrapper.instance().overrideValue('red');
      expect(testCase.props.onChange).toHaveBeenCalledTimes(1);
      expect(testCase.props.onChange).toHaveBeenCalledWith({
        h: 0,
        s: 100,
        l: 50,
        r: 255,
        g: 0,
        b: 0,
        a: 1,
        hexInput: false,
        hex: 'ff0000',
        mode: 'hsl',
        channel: 'h'
      });
      expect(wrapper.instance().state.initialValue).toEqual('#000');
    });

    test('overrideValue with true as second argument sets overrides initialValue', () => {
      wrapper.instance().overrideValue('red', true);
      expect(testCase.props.onChange).toHaveBeenCalledTimes(1);
      expect(testCase.props.onChange).toHaveBeenCalledWith({
        h: 0,
        s: 100,
        l: 50,
        r: 255,
        g: 0,
        b: 0,
        a: 1,
        hexInput: false,
        hex: 'ff0000',
        mode: 'hsl',
        channel: 'h'
      });
      expect(wrapper.instance().state.initialValue).toEqual('red');
    });
  });

  describe(testCases.readOnly.description, () => {
    beforeEach(() => {
      testCase = testCases.readOnly;
      wrapper = shallow(React.createElement(testCase.component, testCase.props));
    });

    test('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe(testCases.hexValue.description, () => {
    beforeEach(() => {
      testCase = testCases.hexValue;
      wrapper = shallow(React.createElement(testCase.component, testCase.props));
    });

    test('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('hex value remains long', () => {
      const value = wrapper.find('[data-test="hex-input"]').props().value;
      expect(value).toEqual('33ffee');
    });
  });

  describe(testCases.shortHexValue.description, () => {
    beforeEach(() => {
      testCase = testCases.shortHexValue;
      wrapper = shallow(React.createElement(testCase.component, testCase.props));
    });

    test('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('hex value remains short', () => {
      const value = wrapper.find('[data-test="hex-input"]').props().value;
      expect(value).toEqual('3fe');
    });
  });

  describe(testCases.rgbValue.description, () => {
    beforeEach(() => {
      testCase = testCases.rgbValue;
      wrapper = shallow(React.createElement(testCase.component, testCase.props));
    });

    test('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('onChange with correct values', () => {
      const mockEvent = {
        target: {
          value: 'rgb'
        }
      };
      wrapper.find('[data-test="mode-rgb"]').props().onChange(mockEvent);
      wrapper.update();
      expect(testCase.props.onChange).toHaveBeenCalledTimes(1);
      expect(testCase.props.onChange).toHaveBeenCalledWith({
        h: 180,
        s: 100,
        l: 50,
        r: 0,
        g: 255,
        b: 255,
        a: 0.5,
        hexInput: false,
        hex: '00ffff',
        mode: 'rgb',
        channel: 'h'
      });
    });
  });

  describe(testCases.hslValue.description, () => {
    beforeEach(() => {
      testCase = testCases.hslValue;
      wrapper = shallow(React.createElement(testCase.component, testCase.props));
    });

    test('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe(testCases.hslaValue.description, () => {
    beforeEach(() => {
      testCase = testCases.hslaValue;
      wrapper = shallow(React.createElement(testCase.component, testCase.props));
    });

    test('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('onChange with correct values', () => {
      const mockEvent = {
        target: {
          value: 'hsl'
        }
      };
      wrapper.find('[data-test="mode-hsl"]').props().onChange(mockEvent);
      wrapper.update();
      expect(testCase.props.onChange).toHaveBeenCalledTimes(1);
      expect(testCase.props.onChange).toHaveBeenCalledWith({
        h: 180,
        s: 100,
        l: 50,
        r: 0,
        g: 255,
        b: 255,
        a: 0.5,
        hexInput: false,
        hex: '00ffff',
        mode: 'hsl',
        channel: 'h'
      });
    });
  });

  describe(testCases.allOptions.description, () => {
    beforeEach(() => {
      testCase = testCases.allOptions;
      wrapper = shallow(React.createElement(testCase.component, testCase.props));
    });

    test('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('correct mode is active', () => {
      const checked = wrapper.find('[data-test="mode-rgb"]').props().checked;
      expect(checked).toEqual(true);
    });

    test('reset action is not present', () => {
      const reset = wrapper.find('[data-test="color-reset"]').exists();
      expect(reset).toBe(false);
    });
  });
});
