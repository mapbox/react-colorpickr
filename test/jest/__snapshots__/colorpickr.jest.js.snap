// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colorpickr basic renders 1`] = `
<div
  className="colorpickr round inline-block bg-gray-faint p12 txt-s"
>
  <div
    className="flex-parent"
  >
    <div
      className="flex-child z1 w180 h180 relative"
    >
      <RGBGradient
        active={false}
        color="r"
        opacityHigh={Object {}}
        opacityLow={Object {}}
        theme={
          Object {
            "gradient": "absolute top right bottom left",
            "gradientBHigh": "gradient-rgb gradient-b-high",
            "gradientBLow": undefined,
            "gradientGHigh": "gradient-rgb gradient-g-high",
            "gradientGLow": "gradient-rgb gradient-g-low",
            "gradientRHigh": "gradient-rgb gradient-r-high",
            "gradientRLow": "gradient-rgb gradient-r-low",
          }
        }
      />
      <RGBGradient
        active={false}
        color="g"
        opacityHigh={Object {}}
        opacityLow={Object {}}
        theme={
          Object {
            "gradient": "absolute top right bottom left",
            "gradientBHigh": "gradient-rgb gradient-b-high",
            "gradientBLow": undefined,
            "gradientGHigh": "gradient-rgb gradient-g-high",
            "gradientGLow": "gradient-rgb gradient-g-low",
            "gradientRHigh": "gradient-rgb gradient-r-high",
            "gradientRLow": "gradient-rgb gradient-r-low",
          }
        }
      />
      <RGBGradient
        active={false}
        color="b"
        opacityHigh={Object {}}
        opacityLow={Object {}}
        theme={
          Object {
            "gradient": "absolute top right bottom left",
            "gradientBHigh": "gradient-rgb gradient-b-high",
            "gradientBLow": undefined,
            "gradientGHigh": "gradient-rgb gradient-g-high",
            "gradientGLow": "gradient-rgb gradient-g-low",
            "gradientRHigh": "gradient-rgb gradient-r-high",
            "gradientRLow": "gradient-rgb gradient-r-low",
          }
        }
      />
      <HGradient
        active={true}
        hueBackground="#002fff"
        theme={
          Object {
            "gradient": "absolute top right bottom left",
            "gradientDarkBottom": "gradient-dark-bottom",
            "gradientLightLeft": "gradient-light-left",
          }
        }
      />
      <SVGradient
        active={false}
        color="s"
        opacityHigh={Object {}}
        opacityLow={Object {}}
        theme={
          Object {
            "gradient": "absolute top right bottom left",
            "gradientDarkBottom": "gradient-dark-bottom",
            "gradientLightBottom": "gradient-light-bottom",
            "gradientSHigh": "gradient-s-high",
            "gradientSLow": "gradient-s-low",
            "gradientVHigh": "gradient-v-high",
            "gradientVLow": "gradient-v-low",
          }
        }
      />
      <SVGradient
        active={false}
        color="v"
        opacityHigh={Object {}}
        opacityLow={Object {}}
        theme={
          Object {
            "gradient": "absolute top right bottom left",
            "gradientDarkBottom": "gradient-dark-bottom",
            "gradientLightBottom": "gradient-light-bottom",
            "gradientSHigh": "gradient-s-high",
            "gradientSLow": "gradient-s-low",
            "gradientVHigh": "gradient-v-high",
            "gradientVLow": "gradient-v-low",
          }
        }
      />
      <XYControl
        isDark="dark"
        onChange={[Function]}
        theme={
          Object {
            "xyControl": "xy-control cursor-move",
            "xyControlContainer": "xy-control-container",
            "xyControlDark": "xy-control-dark",
          }
        }
        x={74}
        xmax={100}
        y={2}
        ymax={100}
      />
      <div
        className="colormode-slider colormode-slider-h"
      >
        <input
          max={359}
          min={0}
          onChange={[Function]}
          style={Object {}}
          type="range"
          value={229}
        />
      </div>
    </div>
    <div
      className="flex-child w120 pl24"
    >
      <div
        className="grid mb12"
      >
        <button
          className="col col--6 btn btn--gray-light py3 round-l is-active"
          data-test="button-mode-rgb"
          onClick={[Function]}
          value="rgb"
        >
          RGB
        </button>
        <button
          className="col col--6 btn btn--gray-light py3 round-r"
          data-test="button-mode-hsv"
          onClick={[Function]}
          value="hsv"
        >
          HSV
        </button>
      </div>
      <div>
        <div
          className="mb3 flex-parent"
        >
          <ModeInput
            checked={false}
            name=""
            onChange={[Function]}
            theme={
              Object {
                "modeInput": "cursor-pointer",
                "modeInputContainer": "flex-child flex-child--no-shrink w24 flex-parent flex-parent--center-cross",
              }
            }
          />
          <RGBInput
            label="R"
            onChange={[Function]}
            theme={
              Object {
                "numberInput": "w-full pl18 input input--s bg-white",
                "numberInputContainer": "flex-child flex-child--grow relative",
                "numberInputLabel": "absolute top left pl6 py3 color-gray-light txt-bold",
              }
            }
            value={66}
          />
        </div>
        <div
          className="mb3 flex-parent"
        >
          <ModeInput
            checked={false}
            name=""
            onChange={[Function]}
            theme={
              Object {
                "modeInput": "cursor-pointer",
                "modeInputContainer": "flex-child flex-child--no-shrink w24 flex-parent flex-parent--center-cross",
              }
            }
          />
          <RGBInput
            label="G"
            onChange={[Function]}
            theme={
              Object {
                "numberInput": "w-full pl18 input input--s bg-white",
                "numberInputContainer": "flex-child flex-child--grow relative",
                "numberInputLabel": "absolute top left pl6 py3 color-gray-light txt-bold",
              }
            }
            value={100}
          />
        </div>
        <div
          className="mb3 flex-parent"
        >
          <ModeInput
            checked={false}
            name=""
            onChange={[Function]}
            theme={
              Object {
                "modeInput": "cursor-pointer",
                "modeInputContainer": "flex-child flex-child--no-shrink w24 flex-parent flex-parent--center-cross",
              }
            }
          />
          <RGBInput
            label="B"
            onChange={[Function]}
            theme={
              Object {
                "numberInput": "w-full pl18 input input--s bg-white",
                "numberInputContainer": "flex-child flex-child--grow relative",
                "numberInputLabel": "absolute top left pl6 py3 color-gray-light txt-bold",
              }
            }
            value={251}
          />
        </div>
      </div>
      <div
        className="mb3 mt12 relative"
      >
        <SVAlphaInput
          label="Î±"
          onChange={[Function]}
          theme={
            Object {
              "numberInput": "w-full pl18 input input--s bg-white",
              "numberInputContainer": "flex-child flex-child--grow relative",
              "numberInputLabel": "absolute top left pl6 py3 color-gray-light txt-bold",
            }
          }
          value={100}
        />
      </div>
      <div
        className="bg-tile"
      >
        <input
          max={100}
          min={0}
          onChange={[Function]}
          style={
            Object {
              "background": "linear-gradient(to right, rgba(66,100,251,0), rgba(66,100,251,1))",
            }
          }
          type="range"
          value={100}
        />
      </div>
    </div>
  </div>
  <div
    className="flex-parent mt6"
  >
    <div
      className="flex-child w180 flex-parent flex-parent--center-cross color-gray"
    >
      New
      <span
        className="bg-tile ml3 inline-block h24 w36 round-l relative"
      >
        <div
          className="w-full h-full round-l absolute"
          style={
            Object {
              "backgroundColor": "rgba(66,100,251,1)",
            }
          }
        />
      </span>
      <div
        className="flex-parent flex-parent--center-cross"
      >
        <span
          className="bg-tile inline-block h24 w36 round-r border-l border--gray-faint relative mr3"
        >
          <button
            className="w-full h-full round-r absolute"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#4264fb",
              }
            }
            title="Reset color"
          />
        </span>
        Current
      </div>
    </div>
    <div
      className="flex-child w120 pl24 align-r"
    >
      <div
        className="relative"
      >
        <label
          className="absolute top left pl6 py3 color-gray-light txt-bold"
        >
          #
        </label>
        <input
          className="w-full pl18 input input--s bg-white"
          onChange={[Function]}
          type="text"
          value="4264fb"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Colorpickr rgbValue renders 1`] = `
<div
  className="colorpickr round inline-block bg-gray-faint p12 txt-s"
>
  <div
    className="flex-parent"
  >
    <div
      className="flex-child z1 w180 h180 relative"
    >
      <RGBGradient
        active={false}
        color="r"
        opacityHigh={Object {}}
        opacityLow={Object {}}
        theme={
          Object {
            "gradient": "absolute top right bottom left",
            "gradientBHigh": "gradient-rgb gradient-b-high",
            "gradientBLow": undefined,
            "gradientGHigh": "gradient-rgb gradient-g-high",
            "gradientGLow": "gradient-rgb gradient-g-low",
            "gradientRHigh": "gradient-rgb gradient-r-high",
            "gradientRLow": "gradient-rgb gradient-r-low",
          }
        }
      />
      <RGBGradient
        active={false}
        color="g"
        opacityHigh={Object {}}
        opacityLow={Object {}}
        theme={
          Object {
            "gradient": "absolute top right bottom left",
            "gradientBHigh": "gradient-rgb gradient-b-high",
            "gradientBLow": undefined,
            "gradientGHigh": "gradient-rgb gradient-g-high",
            "gradientGLow": "gradient-rgb gradient-g-low",
            "gradientRHigh": "gradient-rgb gradient-r-high",
            "gradientRLow": "gradient-rgb gradient-r-low",
          }
        }
      />
      <RGBGradient
        active={false}
        color="b"
        opacityHigh={Object {}}
        opacityLow={Object {}}
        theme={
          Object {
            "gradient": "absolute top right bottom left",
            "gradientBHigh": "gradient-rgb gradient-b-high",
            "gradientBLow": undefined,
            "gradientGHigh": "gradient-rgb gradient-g-high",
            "gradientGLow": "gradient-rgb gradient-g-low",
            "gradientRHigh": "gradient-rgb gradient-r-high",
            "gradientRLow": "gradient-rgb gradient-r-low",
          }
        }
      />
      <HGradient
        active={true}
        hueBackground="#00ffff"
        theme={
          Object {
            "gradient": "absolute top right bottom left",
            "gradientDarkBottom": "gradient-dark-bottom",
            "gradientLightLeft": "gradient-light-left",
          }
        }
      />
      <SVGradient
        active={false}
        color="s"
        opacityHigh={Object {}}
        opacityLow={Object {}}
        theme={
          Object {
            "gradient": "absolute top right bottom left",
            "gradientDarkBottom": "gradient-dark-bottom",
            "gradientLightBottom": "gradient-light-bottom",
            "gradientSHigh": "gradient-s-high",
            "gradientSLow": "gradient-s-low",
            "gradientVHigh": "gradient-v-high",
            "gradientVLow": "gradient-v-low",
          }
        }
      />
      <SVGradient
        active={false}
        color="v"
        opacityHigh={Object {}}
        opacityLow={Object {}}
        theme={
          Object {
            "gradient": "absolute top right bottom left",
            "gradientDarkBottom": "gradient-dark-bottom",
            "gradientLightBottom": "gradient-light-bottom",
            "gradientSHigh": "gradient-s-high",
            "gradientSLow": "gradient-s-low",
            "gradientVHigh": "gradient-v-high",
            "gradientVLow": "gradient-v-low",
          }
        }
      />
      <XYControl
        isDark="dark"
        onChange={[Function]}
        theme={
          Object {
            "xyControl": "xy-control cursor-move",
            "xyControlContainer": "xy-control-container",
            "xyControlDark": "xy-control-dark",
          }
        }
        x={100}
        xmax={100}
        y={0}
        ymax={100}
      />
      <div
        className="colormode-slider colormode-slider-h"
      >
        <input
          max={359}
          min={0}
          onChange={[Function]}
          style={Object {}}
          type="range"
          value={180}
        />
      </div>
    </div>
    <div
      className="flex-child w120 pl24"
    >
      <div
        className="grid mb12"
      >
        <button
          className="col col--6 btn btn--gray-light py3 round-l is-active"
          data-test="button-mode-rgb"
          onClick={[Function]}
          value="rgb"
        >
          RGB
        </button>
        <button
          className="col col--6 btn btn--gray-light py3 round-r"
          data-test="button-mode-hsv"
          onClick={[Function]}
          value="hsv"
        >
          HSV
        </button>
      </div>
      <div>
        <div
          className="mb3 flex-parent"
        >
          <ModeInput
            checked={false}
            name=""
            onChange={[Function]}
            theme={
              Object {
                "modeInput": "cursor-pointer",
                "modeInputContainer": "flex-child flex-child--no-shrink w24 flex-parent flex-parent--center-cross",
              }
            }
          />
          <RGBInput
            label="R"
            onChange={[Function]}
            theme={
              Object {
                "numberInput": "w-full pl18 input input--s bg-white",
                "numberInputContainer": "flex-child flex-child--grow relative",
                "numberInputLabel": "absolute top left pl6 py3 color-gray-light txt-bold",
              }
            }
            value={0}
          />
        </div>
        <div
          className="mb3 flex-parent"
        >
          <ModeInput
            checked={false}
            name=""
            onChange={[Function]}
            theme={
              Object {
                "modeInput": "cursor-pointer",
                "modeInputContainer": "flex-child flex-child--no-shrink w24 flex-parent flex-parent--center-cross",
              }
            }
          />
          <RGBInput
            label="G"
            onChange={[Function]}
            theme={
              Object {
                "numberInput": "w-full pl18 input input--s bg-white",
                "numberInputContainer": "flex-child flex-child--grow relative",
                "numberInputLabel": "absolute top left pl6 py3 color-gray-light txt-bold",
              }
            }
            value={255}
          />
        </div>
        <div
          className="mb3 flex-parent"
        >
          <ModeInput
            checked={false}
            name=""
            onChange={[Function]}
            theme={
              Object {
                "modeInput": "cursor-pointer",
                "modeInputContainer": "flex-child flex-child--no-shrink w24 flex-parent flex-parent--center-cross",
              }
            }
          />
          <RGBInput
            label="B"
            onChange={[Function]}
            theme={
              Object {
                "numberInput": "w-full pl18 input input--s bg-white",
                "numberInputContainer": "flex-child flex-child--grow relative",
                "numberInputLabel": "absolute top left pl6 py3 color-gray-light txt-bold",
              }
            }
            value={255}
          />
        </div>
      </div>
      <div
        className="mb3 mt12 relative"
      >
        <SVAlphaInput
          label="Î±"
          onChange={[Function]}
          theme={
            Object {
              "numberInput": "w-full pl18 input input--s bg-white",
              "numberInputContainer": "flex-child flex-child--grow relative",
              "numberInputLabel": "absolute top left pl6 py3 color-gray-light txt-bold",
            }
          }
          value={50}
        />
      </div>
      <div
        className="bg-tile"
      >
        <input
          max={100}
          min={0}
          onChange={[Function]}
          style={
            Object {
              "background": "linear-gradient(to right, rgba(0,255,255,0), rgba(0,255,255,1))",
            }
          }
          type="range"
          value={50}
        />
      </div>
    </div>
  </div>
  <div
    className="flex-parent mt6"
  >
    <div
      className="flex-child w180 flex-parent flex-parent--center-cross color-gray"
    >
      New
      <span
        className="bg-tile ml3 inline-block h24 w36 round-l relative"
      >
        <div
          className="w-full h-full round-l absolute"
          style={
            Object {
              "backgroundColor": "rgba(0,255,255,0.5)",
            }
          }
        />
      </span>
      <div
        className="flex-parent flex-parent--center-cross"
      >
        <span
          className="bg-tile inline-block h24 w36 round-r border-l border--gray-faint relative mr3"
        >
          <button
            className="w-full h-full round-r absolute"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "rgba(0,255,255,0.5)",
              }
            }
            title="Reset color"
          />
        </span>
        Current
      </div>
    </div>
    <div
      className="flex-child w120 pl24 align-r"
    >
      <div
        className="relative"
      >
        <label
          className="absolute top left pl6 py3 color-gray-light txt-bold"
        >
          #
        </label>
        <input
          className="w-full pl18 input input--s bg-white"
          onChange={[Function]}
          type="text"
          value="0ff"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Colorpickr rgbValue renders 2`] = `
<div
  className="colorpickr round inline-block bg-gray-faint p12 txt-s"
>
  <div
    className="flex-parent"
  >
    <div
      className="flex-child z1 w180 h180 relative"
    >
      <RGBGradient
        active={false}
        color="r"
        opacityHigh={Object {}}
        opacityLow={Object {}}
        theme={
          Object {
            "gradient": "absolute top right bottom left",
            "gradientBHigh": "gradient-rgb gradient-b-high",
            "gradientBLow": undefined,
            "gradientGHigh": "gradient-rgb gradient-g-high",
            "gradientGLow": "gradient-rgb gradient-g-low",
            "gradientRHigh": "gradient-rgb gradient-r-high",
            "gradientRLow": "gradient-rgb gradient-r-low",
          }
        }
      />
      <RGBGradient
        active={false}
        color="g"
        opacityHigh={Object {}}
        opacityLow={Object {}}
        theme={
          Object {
            "gradient": "absolute top right bottom left",
            "gradientBHigh": "gradient-rgb gradient-b-high",
            "gradientBLow": undefined,
            "gradientGHigh": "gradient-rgb gradient-g-high",
            "gradientGLow": "gradient-rgb gradient-g-low",
            "gradientRHigh": "gradient-rgb gradient-r-high",
            "gradientRLow": "gradient-rgb gradient-r-low",
          }
        }
      />
      <RGBGradient
        active={false}
        color="b"
        opacityHigh={Object {}}
        opacityLow={Object {}}
        theme={
          Object {
            "gradient": "absolute top right bottom left",
            "gradientBHigh": "gradient-rgb gradient-b-high",
            "gradientBLow": undefined,
            "gradientGHigh": "gradient-rgb gradient-g-high",
            "gradientGLow": "gradient-rgb gradient-g-low",
            "gradientRHigh": "gradient-rgb gradient-r-high",
            "gradientRLow": "gradient-rgb gradient-r-low",
          }
        }
      />
      <HGradient
        active={true}
        hueBackground="#00ffff"
        theme={
          Object {
            "gradient": "absolute top right bottom left",
            "gradientDarkBottom": "gradient-dark-bottom",
            "gradientLightLeft": "gradient-light-left",
          }
        }
      />
      <SVGradient
        active={false}
        color="s"
        opacityHigh={Object {}}
        opacityLow={Object {}}
        theme={
          Object {
            "gradient": "absolute top right bottom left",
            "gradientDarkBottom": "gradient-dark-bottom",
            "gradientLightBottom": "gradient-light-bottom",
            "gradientSHigh": "gradient-s-high",
            "gradientSLow": "gradient-s-low",
            "gradientVHigh": "gradient-v-high",
            "gradientVLow": "gradient-v-low",
          }
        }
      />
      <SVGradient
        active={false}
        color="v"
        opacityHigh={Object {}}
        opacityLow={Object {}}
        theme={
          Object {
            "gradient": "absolute top right bottom left",
            "gradientDarkBottom": "gradient-dark-bottom",
            "gradientLightBottom": "gradient-light-bottom",
            "gradientSHigh": "gradient-s-high",
            "gradientSLow": "gradient-s-low",
            "gradientVHigh": "gradient-v-high",
            "gradientVLow": "gradient-v-low",
          }
        }
      />
      <XYControl
        isDark="dark"
        onChange={[Function]}
        theme={
          Object {
            "xyControl": "xy-control cursor-move",
            "xyControlContainer": "xy-control-container",
            "xyControlDark": "xy-control-dark",
          }
        }
        x={100}
        xmax={100}
        y={0}
        ymax={100}
      />
      <div
        className="colormode-slider colormode-slider-h"
      >
        <input
          max={359}
          min={0}
          onChange={[Function]}
          style={Object {}}
          type="range"
          value={180}
        />
      </div>
    </div>
    <div
      className="flex-child w120 pl24"
    >
      <div
        className="grid mb12"
      >
        <button
          className="col col--6 btn btn--gray-light py3 round-l is-active"
          data-test="button-mode-rgb"
          onClick={[Function]}
          value="rgb"
        >
          RGB
        </button>
        <button
          className="col col--6 btn btn--gray-light py3 round-r"
          data-test="button-mode-hsv"
          onClick={[Function]}
          value="hsv"
        >
          HSV
        </button>
      </div>
      <div>
        <div
          className="mb3 flex-parent"
        >
          <ModeInput
            checked={false}
            name=""
            onChange={[Function]}
            theme={
              Object {
                "modeInput": "cursor-pointer",
                "modeInputContainer": "flex-child flex-child--no-shrink w24 flex-parent flex-parent--center-cross",
              }
            }
          />
          <RGBInput
            label="R"
            onChange={[Function]}
            theme={
              Object {
                "numberInput": "w-full pl18 input input--s bg-white",
                "numberInputContainer": "flex-child flex-child--grow relative",
                "numberInputLabel": "absolute top left pl6 py3 color-gray-light txt-bold",
              }
            }
            value={0}
          />
        </div>
        <div
          className="mb3 flex-parent"
        >
          <ModeInput
            checked={false}
            name=""
            onChange={[Function]}
            theme={
              Object {
                "modeInput": "cursor-pointer",
                "modeInputContainer": "flex-child flex-child--no-shrink w24 flex-parent flex-parent--center-cross",
              }
            }
          />
          <RGBInput
            label="G"
            onChange={[Function]}
            theme={
              Object {
                "numberInput": "w-full pl18 input input--s bg-white",
                "numberInputContainer": "flex-child flex-child--grow relative",
                "numberInputLabel": "absolute top left pl6 py3 color-gray-light txt-bold",
              }
            }
            value={255}
          />
        </div>
        <div
          className="mb3 flex-parent"
        >
          <ModeInput
            checked={false}
            name=""
            onChange={[Function]}
            theme={
              Object {
                "modeInput": "cursor-pointer",
                "modeInputContainer": "flex-child flex-child--no-shrink w24 flex-parent flex-parent--center-cross",
              }
            }
          />
          <RGBInput
            label="B"
            onChange={[Function]}
            theme={
              Object {
                "numberInput": "w-full pl18 input input--s bg-white",
                "numberInputContainer": "flex-child flex-child--grow relative",
                "numberInputLabel": "absolute top left pl6 py3 color-gray-light txt-bold",
              }
            }
            value={255}
          />
        </div>
      </div>
      <div
        className="mb3 mt12 relative"
      >
        <SVAlphaInput
          label="Î±"
          onChange={[Function]}
          theme={
            Object {
              "numberInput": "w-full pl18 input input--s bg-white",
              "numberInputContainer": "flex-child flex-child--grow relative",
              "numberInputLabel": "absolute top left pl6 py3 color-gray-light txt-bold",
            }
          }
          value={50}
        />
      </div>
      <div
        className="bg-tile"
      >
        <input
          max={100}
          min={0}
          onChange={[Function]}
          style={
            Object {
              "background": "linear-gradient(to right, rgba(0,255,255,0), rgba(0,255,255,1))",
            }
          }
          type="range"
          value={50}
        />
      </div>
    </div>
  </div>
  <div
    className="flex-parent mt6"
  >
    <div
      className="flex-child w180 flex-parent flex-parent--center-cross color-gray"
    >
      New
      <span
        className="bg-tile ml3 inline-block h24 w36 round-l relative"
      >
        <div
          className="w-full h-full round-l absolute"
          style={
            Object {
              "backgroundColor": "rgba(0,255,255,0.5)",
            }
          }
        />
      </span>
      <div
        className="flex-parent flex-parent--center-cross"
      >
        <span
          className="bg-tile inline-block h24 w36 round-r border-l border--gray-faint relative mr3"
        >
          <button
            className="w-full h-full round-r absolute"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "hsla(180, 100%, 50%, 0.5)",
              }
            }
            title="Reset color"
          />
        </span>
        Current
      </div>
    </div>
    <div
      className="flex-child w120 pl24 align-r"
    >
      <div
        className="relative"
      >
        <label
          className="absolute top left pl6 py3 color-gray-light txt-bold"
        >
          #
        </label>
        <input
          className="w-full pl18 input input--s bg-white"
          onChange={[Function]}
          type="text"
          value="0ff"
        />
      </div>
    </div>
  </div>
</div>
`;
